src/App.js

```diff
@@ -11,39 +13,263 @@ class App extends React.Component {
+          No, Exit game
+        </Button>
+      </>
+    );
 
     return (
       <div className="App">
         <header className="App-header">
-          <h3>High Card 🚀</h3>
-          {currCardElems}
-          <br />
-          <button onClick={this.dealCards}>Deal</button>
+          <Box
+            sx={{
+              backgroundColor: "#4aedc4",
+              width: 700,
+              height: 500,
+              color: "black",
+            }}
+          >
+            <Typography style={{ fontFamily: "helvetica", fontSize: 20 }}>
+              <Box
+                sx={{
+                  width: 700,
+                  height: 90,
+                }}
+              >
+                <Box
+                  sx={{
+                    display: "flex",
+
+                    justifyContent: "space-evenly",
+
+                    alignItems: "center",
+                  }}
+                >
+                  <Box>
+                    <p>
+                      {this.state.numMatchesPlayer1Won} -- <b>Player 1</b>
+                    </p>
+                  </Box>
+
+                  <Box>
+                    <p>High Card 🚀</p>
+                    <p>
+                      {" "}
+                      Match: {this.state.matchNumber} --{" "}
+                      {this.state.cardDeck.length} Cards
+                    </p>
+                  </Box>
+
+                  <Box>
+                    {this.state.numMatchesPlayer2Won} -- <b>Player 2</b>
+                  </Box>
+                </Box>
+              </Box>
+              <div style={{ height: 50, width: 700 }}></div>
+              <Box
+                sx={{
+                  width: 700,
+                  height: 150,
+
+                  display: "flex",
+
+                  justifyContent: "space-around",
+                }}
+              >
+                {currCardElems}
+              </Box>
+
+              {this.state.hasGameBegin === true && (
+                <Box
+                  sx={{
+                    display: "flex",
+
+                    justifyContent: "space-evenly",
+                  }}
+                >
+                  <p>{this.state.player1Score}</p>
+                  <p>current scores</p>
+                  <p> {this.state.player2Score}</p>
+                </Box>
+              )}
+
+              {this.state.cardDeck.length > 0 && dealCardsButton}
+              <p>{this.state.hasGameBegin && winnerPerRound}</p>
+              <p>{this.state.cardDeck.length === 0 && winnerPerMatch}</p>
+              <Stack
+                direction="row"
+                spacing={4}
+                style={{
+                  display: "flex",
+                  justifyContent: "center",
+                  alignItems: "center",
+                }}
+              >
+                {this.state.cardDeck.length === 0 && continueGameButton}
+                {this.state.cardDeck.length === 0 && exitGameButton}
```

Comment:
You can summarize similar state lookups into a single one:
```suggestion
                {!this.state.cardDeck.length && (
                <>
                    <Button ...continue game button/>
                    <Button ...exit game button/>
                </>
                )}
```